totalRP3/core/impl/slash.lua

```diff
@@ -46,15 +47,35 @@ function TRP3_API.slash.unregisterCommand(commandID)
 	COMMANDS[commandID] = nil;
 end
 
+
+function TRP3_API.slash.registerConditional(conditionalStructure)
+	assert(conditionalStructure and conditionalStructure.id, "Conditional structure must have and id.");
+	assert(not CONDITIONALS[conditionalStructure.id], "Already registered conditional id: " .. tostring(conditionalStructure.id));
+	CONDITIONALS[conditionalStructure.id] = conditionalStructure;
+
+	--we automatically create the inverse as well
+	CONDITIONALS["no"..conditionalStructure.id] = {
+		id = "no"..conditionalStructure.id,
+		test = function(...)
+			return not conditionalStructure.test(...)
+		end
+	}
+end
+
+function TRP3_API.slash.unregisterConditional(conditionalID)

```

Comment:
This'll be nitpicky and probably can be ignored, but I'm not seeing much of a use case for where someone would want to unregister a conditional.

From a user's perspective it'd be really weird if their macro suddenly stopped working because some addon unregistered itself.

Reply:
This was added under the logic of:
1. it's three lines of code
2. if you can remove slash commands you should be able to remove slash conditionals.

Reply:
I'm okay with it, and as was said on Discord (I think) it might be used in advanced use cases in Extended.